packages/eslint-plugin/src/rules/hook-naming.ts

```diff
@@ -0,0 +1,42 @@
+import { ESLintUtils } from '@typescript-eslint/utils';
+
+import { createRule } from '../utils/createRule';
+import { isIdentifier, isMakeStylesIdentifier } from '../utils/helpers';
+
+export const RULE_NAME = 'hook-naming';
+
+export const hookNamingRule: ReturnType<ReturnType<typeof ESLintUtils.RuleCreator>> = createRule({
+  name: RULE_NAME,
+  meta: {
+    type: 'problem',
+    docs: {
+      description: 'Enforce that hooks returned from makeStyles() calls are named with a "use" prefix',
+      recommended: 'error',
+    },
+    messages: {
+      invalidMakeStylesHookNameFound: 'Hooks must start with the prefix "use"',
```

Comment:
```suggestion
      invalidMakeStylesHookNameFound: '`makeStyles` returns a hook. Hooks must start with the prefix `use`',
```

nit: let's be explicit this lint rule is supposed to target `makeStyles` calls